### PR TITLE
Hook up the batch edit page with our event API, add some logic around making users aware of updates

### PR DESCRIPTION
**Pull request by bp323**

### CODE REVIEW

**Comment by bp323** [comment]
Reassigning to @Coenego to have an initial look. Submitting this PR already as I'd like to agree on the workflow of saving before I put more time in it.

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 1]
```diff
@@ -81,6 +81,108 @@ define(['gh.api.series', 'gh.api.util', 'gh.admin-constants'], function(seriesAP
...
+        } else {
+            // Mark the row so it's visually obvious that an edit was made to it
+            $('.gh-batch-edit-events-container tbody tr[data-eventid="' + $(this).closest('tr').data('eventid') + '"]').removeClass('danger active success').addClass('active');
+            return value;
+        }
+    };
+
+    /**
+     * Set up editable fields in the batch edit tables
+     */
```
Comment (Coenego) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: `@private` needed

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 2]
```diff
@@ -81,6 +81,108 @@ define(['gh.api.series', 'gh.api.util', 'gh.admin-constants'], function(seriesAP
...
+        // Apply jEditable for inline editing
+        $('.jeditable-field').editable(editableSubmitted, {
+            'onblur': 'submit',
+            'select' : true
+        });
+    };
+
+    /**
+     * Submit all changes made in batch edit mode
+     */
```
Comment (Coenego) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: `@private` needed

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 3]
```diff
@@ -81,6 +81,108 @@ define(['gh.api.series', 'gh.api.util', 'gh.admin-constants'], function(seriesAP
...
+                eventObjs.push(eventObj);
+            });
+        });
+
+        var done = 0;
+        var todo = eventObjs.length;
+        var hasError = false;
+
+        /**
+         * Update a single event. Calls itsself when more events need updating and executes
```
Comment (Coenego) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: itself*

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 4]
```diff
@@ -81,6 +81,108 @@ define(['gh.api.series', 'gh.api.util', 'gh.admin-constants'], function(seriesAP
...
+         * @param  {Object}      updatedEvent    The event that needs to be updated
+         * @param  {Function}    callback        Standard callback function
+         * @private
+         */
+        var submitEventUpdate = function(updatedEvent, callback) {
+            eventAPI.updateEvent(updatedEvent.id, updatedEvent.displayName, null, null, null, null, updatedEvent.location, updatedEvent.notes, function(err, data) {
+                if (err) {
+                    hasError = true;
+                    // Mark the row so it's visually obvious that the update failed
+                    $('.gh-batch-edit-events-container tbody tr[data-eventid="' + updatedEvent.id + '"]').addClass('danger');
```
Comment (Coenego) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: Cool, haven't thought of this before!

**Comment by Coenego** [comment]
Looks good, reassigning to @bp323 for some minor follow-up

**Pull request merged**